jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientAuthenticationTest.java

```diff
@@ -689,7 +689,7 @@ public void testTestHeaderInfoParsingUnusualCases() {
         Assert.assertTrue(headerInfo.getType().equalsIgnoreCase("Scheme"));
         Assert.assertTrue(headerInfo.getParameter("realm") == null);
         
-        List<HeaderInfo> headerInfos = aph.getHeaderInfo("Scheme1 ,  Scheme2 ");
+        List<HeaderInfo> headerInfos = aph.getHeaderInfo("Scheme1 ,  Scheme2");
```

Comment:
I would have to also have the test for `Scheme1, Scheme2<SP>`.
I am guessing that when you move the algorithm above to iterate rather than recurse, the current regexp will match.